stats.go

```diff
@@ -155,3 +206,41 @@ func (stats *Stats) socketRefs(delta int) {
 		statsMutex.Unlock()
 	}
 }
+
+func (stats *Stats) noticeSocketAcquisition(waitTime time.Duration) {
+	if stats != nil {
+		statsMutex.Lock()
+		stats.TimesSocketAcquired++
+		stats.TotalPoolWaitTime += waitTime
+		if waitTime > 0 {
+			stats.TimesWaitedForPool++
+		}
+		eventStruct := SocketAcquireMetric{
+			WaitTime: waitTime,
+		}
+		// It's important that the send is non-blocking, since we hold the stats mutex.
+		// If we block, we'll hold up a whole bunch of different mgo operations.
+		select {
+		case statsEvents.OnSocketAcquired <- eventStruct:
+		default:
+		}
+		statsMutex.Unlock()
```

Comment:
Another nitpick: 
You could either defer the mutex unlock, or move if before the send on the channel.

Reply:
The channel send doesn't need the mutex, and there's no particular reason to order the events in the channel according to what order the stats mutex was acquired in (which isn't even guaranteed to be the order mongo saw the connections in) so I moved the unlock above the send.